DSharpPlus.UnifiedCommands/Message/Internals/MessageConditionHandler.cs

```diff
@@ -0,0 +1,29 @@
+using DSharpPlus.UnifiedCommands.Message.Conditions;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DSharpPlus.UnifiedCommands.Message.Internals;
+
+internal class MessageConditionHandler
+{
+    private readonly IReadOnlyList<Func<IServiceProvider, IMessageCondition>> _conditionBuilders;
+
+    internal MessageConditionHandler(IReadOnlyList<Func<IServiceProvider, IMessageCondition>> conditionBuilders)
+        => _conditionBuilders = conditionBuilders;
+
+    internal async Task<bool> StartGoingThroughConditionsAsync(MessageContext context, IServiceScope scope)
```

Comment:
IterateConditionsAsync? also, see my previous comment somewhere about conditions and expression trees